src/rest_vol_attr.c

```diff
@@ -2351,6 +2358,10 @@ RV_attr_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_attr_speci
                                 FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTINC, FAIL,
                                                 "can't increment field's ref. count for copy of attribute's "
                                                 "parent datatype");
+                            if (H5Iinc_ref(loc_obj->u.datatype.tapl_id) < 0)
```

Comment:
We should probably add if blocks here as well to check `loc_obj->u.datatype.tcpl_id != H5P_DATATYPE_CREATE_DEFAULT` and `loc_obj->u.datatype.tapl_id != H5P_DATATYPE_ACCESS_DEFAULT` before incrementing the ref. count for them